### ISSUE TITLE
File tree: add a filter input field similar to the Diff list

### ISSUE THREAD

**Issue by Gibibit** [gitextensions/gitextensions#10157]
### Feature description

Similar to how you can use the "Filter files using a regular expression..." input field in the Diff list of a commit, it would be useful to do the same in the File tree. While a search function exists (can be found with Ctrl+F or right clicking the File tree), it is not quite on par with the filter mode of the Diff list.

A filter field like in the Diff list would be more easily accessible (better UX), and the added functionality would be great for finding files when browsing the history of larger git repositories.

**Advantages**
1. A visible input field makes a common action more accessible to new users, since it is easier to find and does not require the knowledge of a keyboard shortcut or context menu item.
2. Viewing the filter results inside the File tree would allow for clicking through the results without resetting the search results as currently happens with the File tree search popup. Quickly viewing the file contents of files with related names makes it easier to find the required file when the user is unsure of the exact file name.
3. Adding a new filter results view to the File tree is a nice moment to further polish the search by adding file type icons to the results. In contrast to the currently text only search results of the "Find in search tree" popup.

**Points of contention**
1. The File tree is currently a tree view. Adding a filter to the File tree would conflict with this view, since filter results would have to be displayed in a flat view.
2. The File tree search function brings the user to the tree location of the found file, which is useful context. How would this be functionality transferred to a filter view?
4. Since the File tree includes folders, how would they be displayed in a File tree filter view? Should you be able to click them to return to the full tree view? Folders add possible extra complexity to the feature.
5. Would it be desirable for a filter view to supersede the search popup?

### Reference images
![image](https://user-images.githubusercontent.com/1710540/185567953-8f505c1c-d2f0-42bd-ae34-0b526faf2f97.png)
_The Diff filter field referred to as an example. Git Extensions 4.0.0.15025_

![image](https://user-images.githubusercontent.com/1710540/185569943-85294c0b-2172-47a8-81f2-1e398329d5f7.png)
_The file search popup in the File tree displaying the same search query. Double clicking any of the results expands the File tree to the clicked file and selects the file. The search popup and it's results are hidden afterwards. Git Extensions 4.0.0.15025_

Apologies for the lengthy issue description for what may seem a very small thing. In short I do think that replacing the search popup with a filter field would streamline the File tree. Any discussion regarding the usefulness and feasability of the feature is appreciated.

### Environment

- Git Extensions 4.0.0.15025
- Build e8588e4c048e94d62054911fc09decb8139931f6
- Git 2.31.1.windows.1 (recommended: 2.37.1 or later)
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.7
- DPI 168dpi (175% scaling)

**Comment by gerhardol** [comment]
Maybe a searchbox with dropdown as now similar to the sidepanel but when leaving hiding folders (expanded or not) and files not matching the pattern
This would then replace the popup.

**Comment by RussKie** [comment]
> 1. The File tree is currently a tree view. Adding a filter to the File tree would conflict with this view, since filter results would have to be displayed in a flat view.

I don't think this is necessarily true. It's possible to retain the tree view and only show files matching the filter.

> 4\. Would it be desirable for a filter view to supersede the search popup?

👍 from me :)

**Comment by RussKie** [comment]
FYI, there has been a request to present the diff as a tree view (https://github.com/gitextensions/gitextensions/issues/1326), as arguably it presents a better UX.
